test/ApiCheck.Test/ReportGenerationTests.cs

```diff
@@ -0,0 +1,974 @@
+﻿using System;
+using System.Linq;
+using System.Reflection;
+using ApiCheckBaseline.V2;
+using Scenarios;
+using Xunit;
+
+namespace ApiCheck.Test
+{
+    public class ReportGenerationTests
+    {
+        public Assembly V1Assembly => typeof(ApiCheckBaselineV1).GetTypeInfo().Assembly;
+        public Assembly V2Assembly => typeof(ApiCheckBaselineV2).GetTypeInfo().Assembly;
+
+        [Fact]
+        public void DetectsClasses()
+        {
+            // Arrange
+            var generator = CreateGenerator(V1Assembly);
+
+            // Act
+            var report = generator.GenerateBaseline();
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            Assert.Single(report.Types, t => t.Id == "public class Scenarios.BasicClass");
+        }
+
+        [Fact]
+        public void DetectsStructs()
+        {
+            // Arrange
+            var generator = CreateGenerator(V1Assembly);
+
+            // Act
+            var report = generator.GenerateBaseline();
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            Assert.Single(report.Types, t => t.Id == "public struct Scenarios.BasicStruct");
+        }
+
+        [Fact]
+        public void DetectsDerivedClasses()
+        {
+            // Arrange
+            var generator = CreateGenerator(V1Assembly);
+
+            // Act
+            var report = generator.GenerateBaseline();
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            Assert.Single(report.Types, t => t.Id == "public class Scenarios.DerivedClass : Scenarios.BasicClass");
+        }
+
+        [Fact]
+        public void DetectsBasicInterface()
+        {
+            // Arrange
+            var generator = CreateGenerator(V1Assembly);
+
+            // Act
+            var report = generator.GenerateBaseline();
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            Assert.Single(report.Types, t => t.Id == "public interface Scenarios.IBasicInterface");
+        }
+
+        [Fact]
+        public void DetectsComplexInterface()
+        {
+            // Arrange
+            var generator = CreateGenerator(V1Assembly);
+
+            // Act
+            var report = generator.GenerateBaseline();
+
+            // Assert
+            Assert.NotNull(report);
+            Assert.NotNull(report.Types);
+            Assert.Single(report.Types, t => t.Id == "public interface Scenarios.IComplexInterface : Scenarios.IBasicInterface");
+        }
+
+        [Fact]
+        public void DetectsMultipleLevelInterface()
```

Comment:
It means an interface that is part of a hierarchy of more than one level of interaces, for example:
````C#
public interface A {}
public interface B : A {}
public interface C : B {}
```